### PR TITLE
kcli 99.brew-v1 (new formula)

### PR DESCRIPTION
**Pull request by e-minguez**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/kcli.rb [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    url "https://files.pythonhosted.org/packages/cc/3c/3e8c69cd493297003da83f26ccf1faea5dd7da7892a0a7c965ac3bcba7bf/zipp-3.8.0.tar.gz"
+    sha256 "56bf8aadb83c24db6c4b577e13de374ccfb67da2078beba1d037c17980bf43ad"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    assert_match "version: 99.0 commit", shell_output("#{bin}/kcli version")
```
Comment (carlocab) [inline Formula/kcli.rb:109]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

File: Formula/kcli.rb [hunk 2]
```diff
@@ -0,0 +1,111 @@
+class Kcli < Formula
+  include Language::Python::Virtualenv
+
+  desc "Management tool for libvirt/aws/gcp/kubevirt/openstack/ovirt/vsphere/packet"
+  homepage "https://kcli.readthedocs.io/en/latest/"
+  url "https://github.com/karmab/kcli/archive/326734afa7cd0704a8a95854e7f30686ff51b4f9.tar.gz"
+  sha256 "ba48b5dff2226499b1f8a28f318f95471cb19bf140240f80be9bee74ead8f49b"
```
Comment (chenrui333) [inline Formula/kcli.rb:7]: Can you also kindly ask upstream to do a tagged a release?
```suggestion
  url "https://github.com/karmab/kcli/archive/326734afa7cd0704a8a95854e7f30686ff51b4f9.tar.gz"
  version "99.0.202206301135"
  sha256 "ba48b5dff2226499b1f8a28f318f95471cb19bf140240f80be9bee74ead8f49b"
```

Comment (e-minguez) [reply Formula/kcli.rb:7]: @karmab this is for you :D

Comment (e-minguez) [reply Formula/kcli.rb]: > Can you also kindly ask upstream to do a tagged a release?

They created a tag :) Thanks!

File: Formula/kcli.rb (hunk 1 above)
Comment (e-minguez) [reply Formula/kcli.rb]: I've added a better test. Would you mind taking a look at it again? Thanks!

File: Formula/kcli.rb [hunk 3]
```diff
@@ -0,0 +1,112 @@
+class Kcli < Formula
+  include Language::Python::Virtualenv
+
+  desc "Management tool for libvirt/aws/gcp/kubevirt/openstack/ovirt/vsphere/packet"
+  homepage "https://kcli.readthedocs.io/en/latest/"
+  url "https://github.com/karmab/kcli/archive/refs/tags/brew-v1.tar.gz"
```
Comment (SMillerDev) [inline Formula/kcli.rb:6]: What is brew specific about this release?

File: Formula/kcli.rb [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  sha256 "ee2f0842805a39d62d764cf7a71036062feaaae855d1c689e20023ab894d556e"
+  license "Apache-2.0"
+  head "https://github.com/karmab/kcli.git", branch: "master"
+
+  depends_on "libvirt"
+  depends_on "python@3.9"
```
Comment (SMillerDev) [inline Formula/kcli.rb:12]: Can it work with 3.10?

Comment (e-minguez) [reply Formula/kcli.rb]: In fact, it doesn't https://github.com/Homebrew/homebrew-core/actions/runs/2608880913

**Comment by e-minguez** [comment]
Is there anything else I can do in order to move this forward? Thanks!

**Pull request closed without merging**